### PR TITLE
XW-2851 | AdBlockEnabled flag in Qualaroo

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-2851

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/Qualaroo/scripts/Qualaroo.js [hunk 1]
```diff
@@ -70,5 +80,17 @@
 		createCookie('qualaroo_survey_submission');
 	});
 
+	if(ads.context.opts.sourcePointDetection) {
+		document.addEventListener('sp.blocking', function () {
```
Comment (kvas-damian) [inline extensions/wikia/Qualaroo/scripts/Qualaroo.js]: Is it possible that this event handler will be registered after event was fired? Can we mitigate that problem?

File: extensions/wikia/Qualaroo/scripts/Qualaroo.js [hunk 2]
```diff
@@ -15,10 +15,20 @@
...
+	function loadQualaroo () {
+		setTimeout(function(){
+			var d = document, f = d.getElementsByTagName('script')[0], s = d.createElement('script'); s.type = 'text/javascript';
+			s.async = true; s.src = window.wgQualarooUrl; f.parentNode.insertBefore(s, f);
+		}, 1);
+	}
+
+	function setAdBlockEnabledAndLoadQualaroo(enabled) {
+		_kiq.push(['set', {
+			'adBlockedEnabled': enabled
```
Comment (rogatty) [inline extensions/wikia/Qualaroo/scripts/Qualaroo.js]: `adBlocked` or `adBlock`?

File: extensions/wikia/Qualaroo/scripts/Qualaroo.js (hunk 1 above)
Comment (rogatty) [reply extensions/wikia/Qualaroo/scripts/Qualaroo.js]: I wonder if this piece of code would fit better in the file that handles SP integration. We could require this file in there or something like that.

File: extensions/wikia/Qualaroo/scripts/Qualaroo.js [hunk 3]
```diff
@@ -70,5 +80,21 @@
 		createCookie('qualaroo_survey_submission');
 	});
 
+	if (ads.context.opts.sourcePointDetection) {
```
Comment (kvas-damian) [inline extensions/wikia/Qualaroo/scripts/Qualaroo.js]: Please decide `ads` vs `window.ads`.

File: extensions/wikia/Qualaroo/scripts/Qualaroo.js [hunk 4]
```diff
@@ -70,5 +80,21 @@
...
+		if (!window.ads || !window.ads.runtime || !window.ads.runtime.sp || window.ads.runtime.sp.blocking === undefined) {
```
Comment (rogatty) [inline extensions/wikia/Qualaroo/scripts/Qualaroo.js]: Seems brittle, would be nice to have an integration test.

**Review by rogatty — APPROVED** [review]

File: extensions/wikia/Qualaroo/scripts/Qualaroo.js [hunk 5]
```diff
@@ -70,5 +80,21 @@
 		createCookie('qualaroo_survey_submission');
 	});
 
+	if (window.ads.context.opts.sourcePointDetection) {
+		if (!window.ads.runtime || !window.ads.runtime.sp || window.ads.runtime.sp.blocking === undefined) {
```
Comment (dianafa) [inline extensions/wikia/Qualaroo/scripts/Qualaroo.js:84]: per discussion in the corridor, we can add a comment that this is the same logic as in: https://github.com/Wikia/app/blob/dev/extensions/wikia/ARecoveryEngine/js/sourcePointHelper.js#L54

Comment (latata) [reply extensions/wikia/Qualaroo/scripts/Qualaroo.js:84]: Actually it's not the same; firstly we check if sourcePointDetection is enabled not sourcePointRecovery, secondly we check if blocking is undefined (so not set to false or true).

**Review by hakubo — APPROVED** [review]

**Pull request merged**